### PR TITLE
Ampers - Sara S - Hotel

### PR DESCRIPTION
**Pull request by CheerOnMars**
# Hotel
Congratulations! You're submitting your assignment!

## Comprehension Questions
Question | Answer
:------------- | :-------------
Describe a design decision you had to make when working on this project. What options were you considering? What helped you make your final decision? | I had to decide how to group methods into blocks. I would like to create a Block Class in order to move towards classes with fewer responsibility, but I decided to wait until the refactor portion so I could make sure I could get the methods to work first. 
Describe a concept that you gained more clarity on as you worked on this assignment. | 
Describe a _nominal test_ that you wrote for this assignment. | For the test for the calculating reservation cost method, I made sure that a reservation of 5 nights would cost $1000. 
Describe an _edge case test_ that you wrote for this assignment. | One edge case was putting in a reservation end date that was before the reservation start date. 
How do you feel you did in writing pseudocode first, then writing the tests and then the code? | I thought it was helpful to write the pseudocode first because it helped me keep the classes, variables, and methods straight. It's easy to lose track, but the pseudocode was a useful reference sheet.

### CODE REVIEW

File: lib/hotel_admin.rb [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+      @reservations = []
+      @blocks = []
+      @holds_reservations = []
+      @rooms_held = []
+    end
+
+    def add_reservation(reservation)
+      #note: this is tightly coupled, I think.
+      @reservations << reservation
+      @reservation = reservation
```
Comment (kariabancroft) [inline lib/hotel_admin.rb:20]: Why set this as an instance variable here?

File: lib/hotel_admin.rb [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+      return @reservation
+    end
+
+    def calc_reservations(reservation)
+      reservation.price_night * reservation.calc_duration
```
Comment (kariabancroft) [inline lib/hotel_admin.rb:25]: I think this logic would be better inside the reservation object

File: lib/hotel_admin.rb [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+    end
+
+    def reserve_available_room(checkin_request, checkout_request)
+      avail_room = view_available_rooms(checkin_request, checkout_request)[0]
+      add_reservation(Hotel::Reservation.new({checkin_date: checkin_request, checkout_date: checkout_request, room_id: avail_room}))
+    end
+
+    def create_block(input)
+
+      if input[:block_start_date] == nil || input[:block_end_date] == nil
```
Comment (kariabancroft) [inline lib/hotel_admin.rb:58]: All of this data validation should be in an `initialize` somewhere

**Comment by kariabancroft** [comment]
# Hotel
## What We're Looking For

Feature | Feedback
---     | ---
**Design** |
Demonstrated classes having a single responsibility |Room for improvement - it is easy to fill up the `HotelAdmin` class with lots of logic that doesn't need to be there. I think you have a few instances of this, so keep this in mind. 
Demonstrated loose coupling | Yes - though the `HotelAdmin` is doing some things that it doesn't need to do, I think you generally do try to keep the separation between objects. 
Methods demonstrate a good use of encapsulation, inputs and outputs | Yes
Wave 1 requirements |Yes
Wave 2 requirements |Yes
Wave 3 requirements |Yes
